src/sysv/systemd/sssd-ifp.service.in

```diff
@@ -9,6 +9,9 @@ Environment=DEBUG_LOGGER=--logger=files
 EnvironmentFile=-@environment_file@
 Type=dbus
 BusName=org.freedesktop.sssd.infopipe
-ExecStart=@ifp_exec_cmd@ ${DEBUG_LOGGER}
+ExecStartPre=-/bin/chown @SSSD_USER@:@SSSD_USER@ @logpath@/sssd_ifp.log
+ExecStart=@libexecdir@/sssd/sssd_ifp ${DEBUG_LOGGER} --dbus-activated
 CapabilityBoundingSet= @additional_caps@ CAP_IPC_LOCK CAP_CHOWN CAP_DAC_READ_SEARCH CAP_FOWNER CAP_SETGID CAP_SETUID
```

Comment:
Hi,

in my testing it looks like the `CapabilityBoundingSet` is only needed for the `chown` in `ExecStartPre`. What do you think about adding `+` to `ExecStartPre` and remove `CapabilityBoundingSet` so that `chown` runs with full privileges but `sssd_ifp` has no extra rights?

bye,
Sumit

Reply:
Right, there are a lot of excessive caps listed.
I'm working on clearing this stuff in #6873, 53c49f52a97e103c5d28ea9b2829e147d5e61422 in particular.

This regards of CAP_DAC_OVERRIDE it's a bit tricky because if log file is missing at startup (i.e. nothing to `chown()`) then if sssd_ifp is run under 'root' it needs CAP_DAC_OVERRIDE to create log file in /var/log/sssd
Perhaps this can be worked around by addition of 'touch' to `ExecStartPre`
Moreover, if User=sssd then no `CapabilityBoundingSet` is needed at all (and it's presence is confusing).

I hope to improve all this in #6873

Reply:
Btw, merely removing `CapabilityBoundingSet` means "unlimited capabilities". So to run 'sssd_ifp' under root but without caps it's required to assign empty set, IIUC.

Reply:
Another case where a responder needs CAP_DAC_OVERRIDE:
 - sssd.conf::user=sssd, so sbus server sockets are sssd:sssd owned:
```
srw-------. 1 sssd sssd  0 Aug 25 11:53 sbus-dp_files.234537
srw-------. 1 sssd sssd  0 Aug 25 11:53 sbus-monitor
```
 - but a socket activated responder is run under root for some reason (for example, for sssd_nss this is the only option curretly)
Then a responder needs CAP_DAC_OVERRIDE to connect to monitor and DP (this is a reverse of a problem where sssd_be is run under root but a responder under sssd).
Currently this works because sssd-nss.service doesn't define any `CapabilityBoundingSet` at all, so no limits.
Our options are:
 - make socket activated sssd_nss to be able to run under sssd and then say we don't support mix of root/sssd at runtime
 - keep CAP_DAC_OVERRIDE
 - make sbus sockets sssd:root and rw-rw---
maybe something else.

@sumit-bose, what would you say?

Reply:
Actually:
https://github.com/SSSD/sssd/blob/15a22136e19f192c03758c21fa8e48697fa16857/src/responder/nss/nsssrv.c#L617

so I think another option could be to make sbus server sockets `g+rw`.

Reply:
What I can't understand - what was the source of 
```   
    type=AVC msg=audit(1543524888.125:1495): avc:  denied  { fsetid } for
    pid=7706 comm="sssd_nss" capability=4  scontext=system_u:system_r:sssd_t:s0
    tcontext=system_u:system_r:sssd_t:s0 tclass=capability
```
mentioned in the commit message of 61e4ba58934b20a950255e05797aca25aadc1242

First of all, IIRC, by that time all SSSD components run under root without any capabilities restrictions.
Another thing is that I don't understand how writing to non-executable mem-cache file would trigger such AVC (fsetid).
Moreover, tclass=capability - not a file...

Reply:
And right, I missed '+' before `ExecStartPre`.
In other service files it's solved with deprecated 'PermissionsStartOnly':
```
systemd/sssd-ssh.service.in:19:PermissionsStartOnly=true
systemd/sssd-autofs.service.in:19:PermissionsStartOnly=true
systemd/sssd-sudo.service.in:19:PermissionsStartOnly=true
systemd/sssd-pam.service.in:19:PermissionsStartOnly=true
systemd/sssd-pac.service.in:19:PermissionsStartOnly=true
```
 - it defaults to 'false'.
So currently `ExecStartPre` for 'ifp.service' is executed with "User=..." and doesn't make sense.
I'll fix this in #6873